File: py/cpu.py
from komponen import Komponen

class Cpu(Komponen):
    def __init__(self, merk: str, nama: str, jumlah_core: int, kecepatan_ghz: float):
        super().__init__(merk, nama)
        self.jumlah_core = jumlah_core
        self.kecepatan_ghz = kecepatan_ghz

    def get_jumlah_core(self):
        return self.jumlah_core

    def get_kecepatan_ghz(self):
        return self.kecepatan_ghz

File: java/Cpu.java
public class Cpu extends Komponen {
    private int jumlahCore;
    private double kecepatanGHz;

    public Cpu(String merk, String nama, int jumlahCore, double kecepatanGHz) {
        super(merk, nama);
        this.jumlahCore = jumlahCore;
        this.kecepatanGHz = kecepatanGHz;
    }

    public int getJumlahCore() {
        return jumlahCore;
    }

    public double getKecepatanGHz() {
        return kecepatanGHz;
    }

    public void setJumlahCore(int jumlahCore) {
        this.jumlahCore = jumlahCore;
    }

    public void setKecepatanGHz(double kecepatanGHz) {
        this.kecepatanGHz = kecepatanGHz;
    }

    @Override
    public String toString() {
        return super.toString() + " | " + jumlahCore + " core | " + kecepatanGHz + " GHz";
    }
}
